src/dmd/dsymbolsem.d

```diff
@@ -333,6 +335,27 @@ private FuncDeclaration buildPostBlit(StructDeclaration sd, Scope* sc)
         sd.members.push(_alias);
         _alias.addMember(sc, sd); // add to symbol table
     }
+
+    if (sd.hasCopyCtor)
+    {
+        // we have user defined postblit, so we prioritize it
+        if (hasUserDefinedPosblit)
+        {
+            sd.hasCopyCtor = false;
+            return xpostblit;
+        }
+        // we have fields with postblits, so print deprecations
+        if (xpostblit && !xpostblit.isDisabled())
+        {
+            deprecation(sd.loc, "`struct %s` has both a user defined copy constructor and fields with postblits.", sd.toChars);
+            deprecationSupplemental(sd.loc, "The field postblit will have priority over the copy constructor.");
+            deprecationSupplemental(sd.loc, "To change this, the postblit should be disabled for `struct %s`", sd.toChars());
```

Comment:
```suggestion
            deprecationSupplemental(sd.loc, "Use `@disable this(this);`, define postblit explicitly, or remove or delete the copy constructor");
```

Perhaps also point to which fields trigger the postblit generation, if that's available ?

Reply:
I thought about that, but since the fix is to disable the postblit I think it is not relevant to output the fields.